### PR TITLE
ROX-20134: Close transactions

### PR DESCRIPTION
**Pull request by md2119**
## Description
Close read transactions.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### CODE REVIEW

File: central/node/datastore/store/postgres/store.go [hunk 1]
```diff
@@ -860,6 +872,10 @@ func (s *storeImpl) retryableGetMany(ctx context.Context, ids []string) ([]*stor
 	resultsByID := make(map[string]*storage.Node)
 	for _, id := range ids {
 		msg, found, err := s.getFullNode(ctx, tx, id)
+		// No changes are made to the database, so COMMIT or ROLLBACK have same effect.
+		if err := tx.Commit(ctx); err != nil {
```
Comment (connorgorman) [inline central/node/datastore/store/postgres/store.go:876]: This seems like we don't want to commit here. We have the commit below at the bottom

File: central/image/datastore/store/postgres/store.go [hunk 2]
```diff
@@ -1107,6 +1118,9 @@ func (s *storeImpl) retryableGetMany(ctx context.Context, ids []string) ([]*stor
 	resultsByID := make(map[string]*storage.Image)
 	for _, id := range ids {
 		msg, found, err := s.getFullImage(ctx, tx, id)
+		if err := tx.Commit(ctx); err != nil {
```
Comment (dashrews78) [inline central/image/datastore/store/postgres/store.go:1121]: this goes with @connorgorman comment on the node file.  I think we want this outside the loop for consistency and do all the queries under the same transaction.

File: central/node/datastore/store/postgres/store.go (hunk 1 above)
Comment (dashrews78) [reply central/node/datastore/store/postgres/store.go:876]: CI has answered this question.  Commit closes the transaction so next call fails.

Comment (md2119) [reply central/node/datastore/store/postgres/store.go:876]: Ah, the tx needs to be committed only if `err != nil`.

**Review by dashrews78 — APPROVED** [review]

**Pull request merged**